### PR TITLE
Fixed refit_full using incorrect base models at L2+

### PR DESCRIPTION
**Pull request by Innixma**
Code cleanup

*Issue #, if available:*

*Description of changes:*

- Fixed refit_full using incorrect base models at L2+. The refit_full models were using all ancestors as base models instead of just the immediate predecessors. This issue didn't affect weighted ensembles, so it only occurred if `stack_ensemble_levels>=2` and `refit_full=True`.
- Code cleanup in trainer

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-792-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-792/1/index.html

File: tabular/src/autogluon/tabular/trainer/abstract_trainer.py [hunk 1]
```diff
@@ -212,8 +212,7 @@ def train(self, X_train, y_train, X_val=None, y_val=None, **kwargs):
     # TODO: Remove name_suffix, hacked in
     # TODO: Enable easier re-mapping of trained models -> hyperparameters input (They don't share a key since name can change)
     def train_multi_levels(self, X_train, y_train, X_val, y_val, X_unlabeled=None, hyperparameters: dict = None, base_model_names: List[str] = None,
-                           hyperparameter_tune=False, feature_prune=False, core_kwargs: dict = None, level_start=0, level_end=0,
-                           stack_name_core='core', stack_name_aux='aux1', time_limit=None, name_suffix='') -> List[str]:
+                           hyperparameter_tune=False, feature_prune=False, core_kwargs: dict = None, aux_kwargs: dict = None, level_start=0, level_end=0, time_limit=None, name_suffix='') -> List[str]:
         """
         Trains a multi-layer stack ensemble using the input data on the hyperparameters dict input.
```
Comment (jwmueller) [inline tabular/src/autogluon/tabular/trainer/abstract_trainer.py:217]: docstring should clarify how `core_kwargs, aux_kwargs` are passed to core/aux models.

**Review by jwmueller — APPROVED** [review]
looks good

File: tabular/src/autogluon/tabular/trainer/abstract_trainer.py (hunk 1 above)
Comment (Innixma) [reply tabular/src/autogluon/tabular/trainer/abstract_trainer.py:217]: will write docstrings in a future PR, currently a lot of functionality is being updated.

**Pull request merged**